pkg/tso/allocator_manager.go

```diff
@@ -143,6 +143,16 @@ type ElectionMember interface {
 	PrecheckLeader() error
 }
 
+// ConfigProvider is used to provide TSO configuration.
+type ConfigProvider interface {
```

Comment:
It seems that there is no need to define a new interface. The newly added interface tso.ServiceConfig in pkg/tso/config.go was designed for this purpose. It's already used by tso.KeyspaceGroupManager and planned to use it for simplifying the interface of NewAllocatorManager too.

pd server config.Config just needs to implement  tso.ServiceConfig, i.e., only needs to add the following methods (the other methods have already been implemented, and GeBackendEndpoints() can be removed).

// GetName returns the Name
func (c *Config) GetName() string {
	return c.Name
}

// GetListenAddr returns the ListenAddr
func (c *Config) GetListenAddr() string {
	return c.ClientUrls
}

// GetAdvertiseListenAddr returns the AdvertiseListenAddr
func (c *Config) GetAdvertiseListenAddr() string {
	return c.AdvertiseClientUrls
}


// GetMaxResetTSGap returns the MaxResetTSGap.
func (c *Config) GetMaxResetTSGap() time.Duration {
	return ...
}

Reply:
I just noticed that, for PD/API GetMaxResetTSGap() needs to return s.persistOptions.GetMaxResetTSGap(). Yes, you're right to let Server implement the interface. Shall we combine ConfigProvider and tso.ServiceConfig into one interface? I'm ok to either way.

Reply:
Sorry for merging this PR without noticing your comment. I filed a new PR to address this problem. PTAL.

https://github.com/tikv/pd/pull/6272